### PR TITLE
Coupons + Promotion code api

### PR DESCRIPTION
**Pull request by jramosg**
Coupons + Promotion code api

### CODE REVIEW

File: test/dev/gethop/payments/stripe/coupons_test.clj [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            [dev.gethop.payments.stripe]
+            [dev.gethop.payments.stripe.test-utils :as test-utils]))
+
+(def ^:private ^:const coupon-to-create
+  {:duration "repeating"
+   :duration_in_months 3
+   :name "Test coupon"
+   :percent_off 20.0})
+
+(deftest ^:integration crete-coupon
```
Comment (theodor-alin-meresescu) [inline test/dev/gethop/payments/stripe/coupons_test.clj]: I noticed a small typo in the name here

File: test/dev/gethop/payments/stripe/promotion_code_test.clj [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  (:require [clojure.string :as str]
+            [clojure.test :refer :all]
+            [dev.gethop.payments.core :as core]
+            [dev.gethop.payments.stripe]
+            [dev.gethop.payments.stripe.test-utils :as test-utils]))
+
+(deftest ^:integration create-promotion-code
+  (let [payment-adapter (test-utils/init-payment-adapter)]
+    (testing "Create valid promotion code"
+      (let [coupon-id (test-utils/create-coupon+get-id payment-adapter)
```
Comment (theodor-alin-meresescu) [inline test/dev/gethop/payments/stripe/promotion_code_test.clj:15]: Unless I'm missing something, I don't see an implementation for test-utils/create-coupon+get-id

Comment (jramosg) [reply test/dev/gethop/payments/stripe/promotion_code_test.clj:15]: I missed dev.gethop.payments.stripe.test-utils in the commit, sorry! I will added as soon as I can

Comment (jramosg) [reply test/dev/gethop/payments/stripe/promotion_code_test.clj:15]: changed

File: test/dev/gethop/payments/stripe/coupons_test.clj (hunk 1 above)
Comment (jramosg) [reply test/dev/gethop/payments/stripe/coupons_test.clj]: changed

**Comment by theodor-alin-meresescu** [comment]
Awesome! Thanks for putting this together, Jon

**Pull request merged**